pkg/controller/attach_detach.go

```diff
@@ -113,6 +113,11 @@ func (r *ReconcileAttachDetach) Reconcile(ctx context.Context, request reconcile
 		if err := r.removeFinalizer(ctx, azVolumeAttachment); err != nil {
 			return reconcileReturnOnError(ctx, azVolumeAttachment, "delete", err, r.retryInfo)
 		}
+		// deletion of azVolumeAttachment is succeeded, the node's remaining capacity of disk attachment should be increased by 1
+		remainingCapacity, nodeExists := r.SharedState.nodeDiskAvailabilityMap.Load(azVolumeAttachment.Spec.NodeName)
```

Comment:
You can omit the explicit use of `SharedState` since it's an anonymous field, i.e. `r.nodeDiskAvailabilityMap`.

Consider encapsulating the allocation and deallocation of the attachment count in methods of `SharedState`, e.g.

```go
func (s *SharedState) allocateAttachment(node string) error
func (s* SharedState) deallocateAttachment(node string)
```

Reply:
Sure. Will update and fix the failed unit test together.

Reply:
Done.